fvtest/compilertest/compile/Method.cpp

```diff
@@ -67,8 +67,10 @@ ResolvedMethod::signature(TR_Memory * trMemory, TR_AllocationKind allocKind)
    {
    if( !_signature )
       {
-      char * s = (char *)trMemory->allocateMemory(strlen(_fileName) + 1 + strlen(_lineNumber) + 1 + strlen(_name) + 1, allocKind);
-      sprintf(s, "%s:%s:%s", _fileName, _lineNumber, _name);
+      char * s = (char *)trMemory->allocateMemory(_fileName.length() + 1 + strlen(_lineNumber) + 1 + _name.length() + 1, allocKind);
+
+      // TODO: Change this to use C++ string concatenation
+      sprintf(s, "%s:%s:%s", _fileName.c_str(), _lineNumber, _name.c_str());
```

Comment:
I don't like TODOs.  Presumably, doing this in std:strings and then converting the result to a c_str would eliminate the explicit allocation above?  It would look cleaner I suppose, but let's reason about its efficiency.